x-pack/plugins/endpoint/public/embeddables/resolver/store/data/selectors.ts

```diff
@@ -0,0 +1,302 @@
+/*
```

Comment:
I think we could make this logic easier to read/understand once we've added some tests. Maybe we do it like:

1. port it over (you did this)
2. test it
3. refactor it until its clean and nice while keeping tests passing